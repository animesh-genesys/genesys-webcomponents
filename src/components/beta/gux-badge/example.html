<h1>gux-badge-beta</h1>
<h2>Badges</h2>

<h2>Subtle</h2>
<div class="badges">
  <gux-badge-beta>No Impact</gux-badge-beta>
  <gux-badge-beta color="neutral">No Impact (explicit)</gux-badge-beta>
  <gux-badge-beta color="green">Resolved</gux-badge-beta>
  <gux-badge-beta color="yellow">Medium Impact</gux-badge-beta>
  <gux-badge-beta color="red">High Impact</gux-badge-beta>
</div>

<h2>Bold</h2>
<div class="badges">
  <gux-badge-beta bold color="neutral">No Impact</gux-badge-beta>
  <gux-badge-beta bold color="green">Resolved</gux-badge-beta>
  <gux-badge-beta bold color="yellow">Medium Impact</gux-badge-beta>
  <gux-badge-beta bold color="red">High Impact</gux-badge-beta>
</div>

<h2>Icon + Label</h2>
<div class="badges">
  <gux-badge-beta color="neutral">
    <gux-icon icon-name="subtract" decorative></gux-icon>No Impact
  </gux-badge-beta>
  <gux-badge-beta bold color="neutral">
    <gux-icon icon-name="subtract" decorative></gux-icon>No Impact
  </gux-badge-beta>
</div>

<style>
  .badges {
    display: flex;
    flex-wrap: wrap;
    padding: 10px;
    border: 1px solid var(--gux-charcoal-grey);
  }

  gux-badge-beta {
    max-width: 100px;
    margin: 4px;
  }
</style>
