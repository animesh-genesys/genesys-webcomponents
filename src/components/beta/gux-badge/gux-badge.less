@import (reference) '../../../style/color.less';
@import (reference) '../../../style/mixins.less';
@import (reference) '../../../style/spacing.less';
@import (reference) '../../../style/typography.less';

// Style
:host {
  display: inline-block;
}

.gux-badge {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-content: stretch;
  align-items: center;
  justify-content: flex-start;
  height: 20px;
  padding: @spacing-3xs @spacing-xs;
  color: @gux-grey-100;
  background-color: @gux-black-50;
  border-radius: @spacing-2xs;
  .gux-label();

  gux-tooltip-title {
    white-space: nowrap;
    cursor: default;

    ::slotted(gux-icon) {
      height: 16px;
      font-size: 16px;
    }
  }

  .gux-sr-only {
    .gux-sr-only-clip();
  }

  &.gux-neutral {
    color: @gux-black-50;
    background-color: @gux-grey-60;

    &.gux-bold {
      color: @gux-grey-100;
      background-color: @gux-black-50;
    }
  }

  &.gux-red {
    color: @gux-alert-red-30;
    background-color: @gux-alert-red-100;

    &.gux-bold {
      color: @gux-grey-100;
      background-color: @gux-alert-red-60;
    }
  }

  &.gux-yellow {
    color: @gux-alert-yellow-30;
    background-color: @gux-alert-yellow-100;

    &.gux-bold {
      color: @gux-black-50;
      background-color: @gux-alert-yellow-60;
    }
  }

  &.gux-green {
    color: @gux-alert-green-30;
    background-color: @gux-alert-green-100;

    &.gux-bold {
      color: @gux-grey-100;
      background-color: @gux-alert-green-60;
    }
  }
}
