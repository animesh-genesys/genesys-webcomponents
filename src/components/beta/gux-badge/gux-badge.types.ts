export type GuxBadgeColor = 'neutral' | 'green' | 'yellow' | 'red';
