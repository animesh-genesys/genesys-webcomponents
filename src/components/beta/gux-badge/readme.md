# gux-badge-beta



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description             | Type                                        | Default     |
| -------- | --------- | ----------------------- | ------------------------------------------- | ----------- |
| `bold`   | `bold`    | Bold badge.             | `boolean`                                   | `false`     |
| `color`  | `color`   | Badge background color. | `"green" \| "neutral" \| "red" \| "yellow"` | `'neutral'` |


## Dependencies

### Depends on

- [gux-tooltip-title](../../stable/gux-tooltip-title)

### Graph
```mermaid
graph TD;
  gux-badge-beta --> gux-tooltip-title
  gux-tooltip-title --> gux-tooltip
  style gux-badge-beta fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
