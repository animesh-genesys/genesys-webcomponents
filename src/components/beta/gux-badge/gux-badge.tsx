import { Component, Element, h, JSX, Prop, State } from '@stencil/core';

import { GuxBadgeColor } from './gux-badge.types';

import { trackComponent } from '../../../usage-tracking';
import { buildI18nForComponent, GetI18nValue } from '../../../i18n';
import translationResources from './i18n/en.json';

@Component({
  styleUrl: 'gux-badge.less',
  tag: 'gux-badge-beta',
  shadow: true
})
export class GuxBadge {
  private i18n: GetI18nValue;

  /**
   * Reference to the host element.
   */
  @Element()
  root: HTMLElement;

  /**
   * Badge background color.
   */
  @Prop()
  color: GuxBadgeColor = 'neutral';

  /**
   * Bold badge.
   */
  @Prop()
  bold: boolean = false;

  @State()
  label: string;

  private onSlotChange(event: Event) {
    const slotAssignedNodes = (
      event.composedPath()[0] as HTMLSlotElement
    ).assignedNodes();
    this.label = slotAssignedNodes
      .map(nodeItem => nodeItem.textContent)
      .join('');
  }

  private renderBadgeTitle(): JSX.Element {
    return (
      <gux-tooltip-title>
        <span>
          <slot
            aria-hidden="true"
            onSlotchange={this.onSlotChange.bind(this)}
          />
        </span>
      </gux-tooltip-title>
    ) as JSX.Element;
  }

  private renderSrText(): JSX.Element {
    return (
      <div class="gux-sr-only">
        {this.i18n(this.getVariant(), {
          label: this.label
        })}
      </div>
    ) as JSX.Element;
  }

  private getVariant(): string {
    return `${this.color}${this.bold ? '-bold' : ''}`;
  }

  async componentWillLoad(): Promise<void> {
    trackComponent(this.root);
    this.i18n = await buildI18nForComponent(this.root, translationResources);
  }

  render(): JSX.Element {
    return (
      <div
        class={{
          'gux-badge': true,
          [`gux-${this.color}`]: true,
          'gux-bold': this.bold
        }}
      >
        {this.renderBadgeTitle()}
        {this.renderSrText()}
      </div>
    ) as JSX.Element;
  }
}
