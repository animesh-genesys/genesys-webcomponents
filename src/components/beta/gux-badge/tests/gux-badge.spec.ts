import { newSpecPage } from '@stencil/core/testing';
import { GuxBadge } from '../gux-badge';

const components = [GuxBadge];
const language = 'en';

describe('gux-badge-beta', () => {
  describe('#render', () => {
    [
      '<gux-badge-beta>default</gux-badge-beta>',
      '<gux-badge-beta color="neutral">default explicit</gux-badge-beta>',
      '<gux-badge-beta color="green">sucess</gux-badge-beta>',
      '<gux-badge-beta color="yellow">warning</gux-badge-beta>',
      '<gux-badge-beta color="red">danger</gux-badge-beta>',
      '<gux-badge-beta bold color="green">success bold</gux-badge-beta>',
      '<gux-badge-beta bold color="yellow">warning bold</gux-badge-beta>',
      '<gux-badge-beta bold color="red">danger bold</gux-badge-beta>',
      '<gux-badge-beta color="yellow"><gux-icon icon-name="subtract" decorative></gux-icon>warning icon</gux-badge-beta>',
      '<gux-badge-beta color="red"><gux-icon icon-name="subtract" decorative></gux-icon>danger icon</gux-badge-beta>',
      '<gux-badge-beta color="yellow"><gux-icon icon-name="subtract" decorative></gux-icon>warning icon</gux-badge-beta>',
      '<gux-badge-beta bold color="yellow"><gux-icon icon-name="subtract" decorative></gux-icon>warning bold icon</gux-badge-beta>',
      '<gux-badge-beta bold color="red"><gux-icon icon-name="subtract" decorative></gux-icon>danger bold icon</gux-badge-beta>',
      '<gux-badge-beta bold color="yellow"><gux-icon icon-name="subtract" decorative></gux-icon>warning bold icon</gux-badge-beta>',
      '<gux-badge-beta bold><gux-icon icon-name="subtract" decorative></gux-icon>default bold</gux-badge-beta>',
      '<gux-badge-beta bold><gux-icon icon-name="subtract" decorative></gux-icon>default bold icon</gux-badge-beta>'
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
