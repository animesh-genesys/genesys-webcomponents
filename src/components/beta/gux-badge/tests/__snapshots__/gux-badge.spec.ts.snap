// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-badge-beta #render should render component as expected (1) 1`] = `
<gux-badge-beta>
  <mock:shadow-root>
    <div class="gux-badge gux-neutral">
      <gux-tooltip-title>
        <span>
          <slot aria-hidden="true"></slot>
        </span>
      </gux-tooltip-title>
      <div class="gux-sr-only">
        badge with label: , color: neutral
      </div>
    </div>
  </mock:shadow-root>
  default
</gux-badge-beta>
`;

exports[`gux-badge-beta #render should render component as expected (2) 1`] = `
<gux-badge-beta color="neutral">
  <mock:shadow-root>
    <div class="gux-badge gux-neutral">
      <gux-tooltip-title>
        <span>
          <slot aria-hidden="true"></slot>
        </span>
      </gux-tooltip-title>
      <div class="gux-sr-only">
        badge with label: , color: neutral
      </div>
    </div>
  </mock:shadow-root>
  default explicit
</gux-badge-beta>
`;

exports[`gux-badge-beta #render should render component as expected (3) 1`] = `
<gux-badge-beta color="green">
  <mock:shadow-root>
    <div class="gux-badge gux-green">
      <gux-tooltip-title>
        <span>
          <slot aria-hidden="true"></slot>
        </span>
      </gux-tooltip-title>
      <div class="gux-sr-only">
        badge with label: , color: green
      </div>
    </div>
  </mock:shadow-root>
  sucess
</gux-badge-beta>
`;

exports[`gux-badge-beta #render should render component as expected (4) 1`] = `
<gux-badge-beta color="yellow">
  <mock:shadow-root>
    <div class="gux-badge gux-yellow">
      <gux-tooltip-title>
        <span>
          <slot aria-hidden="true"></slot>
        </span>
      </gux-tooltip-title>
      <div class="gux-sr-only">
        badge with label: , color: yellow
      </div>
    </div>
  </mock:shadow-root>
  warning
</gux-badge-beta>
`;

exports[`gux-badge-beta #render should render component as expected (5) 1`] = `
<gux-badge-beta color="red">
  <mock:shadow-root>
    <div class="gux-badge gux-red">
      <gux-tooltip-title>
        <span>
          <slot aria-hidden="true"></slot>
        </span>
      </gux-tooltip-title>
      <div class="gux-sr-only">
        badge with label: , color: red
      </div>
    </div>
  </mock:shadow-root>
  danger
</gux-badge-beta>
`;

exports[`gux-badge-beta #render should render component as expected (6) 1`] = `
<gux-badge-beta bold="" color="green">
  <mock:shadow-root>
    <div class="gux-badge gux-bold gux-green">
      <gux-tooltip-title>
        <span>
          <slot aria-hidden="true"></slot>
        </span>
      </gux-tooltip-title>
      <div class="gux-sr-only">
        badge with label: , color: green bold
      </div>
    </div>
  </mock:shadow-root>
  success bold
</gux-badge-beta>
`;

exports[`gux-badge-beta #render should render component as expected (7) 1`] = `
<gux-badge-beta bold="" color="yellow">
  <mock:shadow-root>
    <div class="gux-badge gux-bold gux-yellow">
      <gux-tooltip-title>
        <span>
          <slot aria-hidden="true"></slot>
        </span>
      </gux-tooltip-title>
      <div class="gux-sr-only">
        badge with label: , color: yellow bold
      </div>
    </div>
  </mock:shadow-root>
  warning bold
</gux-badge-beta>
`;

exports[`gux-badge-beta #render should render component as expected (8) 1`] = `
<gux-badge-beta bold="" color="red">
  <mock:shadow-root>
    <div class="gux-badge gux-bold gux-red">
      <gux-tooltip-title>
        <span>
          <slot aria-hidden="true"></slot>
        </span>
      </gux-tooltip-title>
      <div class="gux-sr-only">
        badge with label: , color: red bold
      </div>
    </div>
  </mock:shadow-root>
  danger bold
</gux-badge-beta>
`;

exports[`gux-badge-beta #render should render component as expected (9) 1`] = `
<gux-badge-beta color="yellow">
  <mock:shadow-root>
    <div class="gux-badge gux-yellow">
      <gux-tooltip-title>
        <span>
          <slot aria-hidden="true"></slot>
        </span>
      </gux-tooltip-title>
      <div class="gux-sr-only">
        badge with label: , color: yellow
      </div>
    </div>
  </mock:shadow-root>
  <gux-icon decorative="" icon-name="subtract"></gux-icon>
  warning icon
</gux-badge-beta>
`;

exports[`gux-badge-beta #render should render component as expected (10) 1`] = `
<gux-badge-beta color="red">
  <mock:shadow-root>
    <div class="gux-badge gux-red">
      <gux-tooltip-title>
        <span>
          <slot aria-hidden="true"></slot>
        </span>
      </gux-tooltip-title>
      <div class="gux-sr-only">
        badge with label: , color: red
      </div>
    </div>
  </mock:shadow-root>
  <gux-icon decorative="" icon-name="subtract"></gux-icon>
  danger icon
</gux-badge-beta>
`;

exports[`gux-badge-beta #render should render component as expected (11) 1`] = `
<gux-badge-beta color="yellow">
  <mock:shadow-root>
    <div class="gux-badge gux-yellow">
      <gux-tooltip-title>
        <span>
          <slot aria-hidden="true"></slot>
        </span>
      </gux-tooltip-title>
      <div class="gux-sr-only">
        badge with label: , color: yellow
      </div>
    </div>
  </mock:shadow-root>
  <gux-icon decorative="" icon-name="subtract"></gux-icon>
  warning icon
</gux-badge-beta>
`;

exports[`gux-badge-beta #render should render component as expected (12) 1`] = `
<gux-badge-beta bold="" color="yellow">
  <mock:shadow-root>
    <div class="gux-badge gux-bold gux-yellow">
      <gux-tooltip-title>
        <span>
          <slot aria-hidden="true"></slot>
        </span>
      </gux-tooltip-title>
      <div class="gux-sr-only">
        badge with label: , color: yellow bold
      </div>
    </div>
  </mock:shadow-root>
  <gux-icon decorative="" icon-name="subtract"></gux-icon>
  warning bold icon
</gux-badge-beta>
`;

exports[`gux-badge-beta #render should render component as expected (13) 1`] = `
<gux-badge-beta bold="" color="red">
  <mock:shadow-root>
    <div class="gux-badge gux-bold gux-red">
      <gux-tooltip-title>
        <span>
          <slot aria-hidden="true"></slot>
        </span>
      </gux-tooltip-title>
      <div class="gux-sr-only">
        badge with label: , color: red bold
      </div>
    </div>
  </mock:shadow-root>
  <gux-icon decorative="" icon-name="subtract"></gux-icon>
  danger bold icon
</gux-badge-beta>
`;

exports[`gux-badge-beta #render should render component as expected (14) 1`] = `
<gux-badge-beta bold="" color="yellow">
  <mock:shadow-root>
    <div class="gux-badge gux-bold gux-yellow">
      <gux-tooltip-title>
        <span>
          <slot aria-hidden="true"></slot>
        </span>
      </gux-tooltip-title>
      <div class="gux-sr-only">
        badge with label: , color: yellow bold
      </div>
    </div>
  </mock:shadow-root>
  <gux-icon decorative="" icon-name="subtract"></gux-icon>
  warning bold icon
</gux-badge-beta>
`;

exports[`gux-badge-beta #render should render component as expected (15) 1`] = `
<gux-badge-beta bold="">
  <mock:shadow-root>
    <div class="gux-badge gux-bold gux-neutral">
      <gux-tooltip-title>
        <span>
          <slot aria-hidden="true"></slot>
        </span>
      </gux-tooltip-title>
      <div class="gux-sr-only">
        badge with label: , color: neutral bold
      </div>
    </div>
  </mock:shadow-root>
  <gux-icon decorative="" icon-name="subtract"></gux-icon>
  default bold
</gux-badge-beta>
`;

exports[`gux-badge-beta #render should render component as expected (16) 1`] = `
<gux-badge-beta bold="">
  <mock:shadow-root>
    <div class="gux-badge gux-bold gux-neutral">
      <gux-tooltip-title>
        <span>
          <slot aria-hidden="true"></slot>
        </span>
      </gux-tooltip-title>
      <div class="gux-sr-only">
        badge with label: , color: neutral bold
      </div>
    </div>
  </mock:shadow-root>
  <gux-icon decorative="" icon-name="subtract"></gux-icon>
  default bold icon
</gux-badge-beta>
`;
